yt_dlp/extractor/hotstar.py

```diff
@@ -96,56 +110,125 @@ class HotStarIE(HotStarBaseIE):
+            'season_id': 6771,
+            'episode': 'Janhvi Targets Suman',
+            'episode_number': 8,
+        },
+    }, {
+        # contentData
+        'url': 'hotstar:episode:1000234847',
+        'info_dict': {
+            'id': '1000234847',
+            'ext': 'mp4',
+            'title': 'Janhvi Targets Suman',
+            'description': 'md5:78a85509348910bd1ca31be898c5796b',
+            'timestamp': 1556670600,
+            'upload_date': '20190501',
+            'duration': 1219,
+            'channel': 'StarPlus',
+            'channel_id': 3,
+            'series': 'Ek Bhram - Sarvagun Sampanna',
+            'season': 'Chapter 1',
+            'season_number': 1,
+            'season_id': 6771,
+            'episode': 'Janhvi Targets Suman',
+            'episode_number': 8,
+        },
     }]
     _GEO_BYPASS = False
 
     def _real_extract(self, url):
-        video_id = self._match_id(url)
+        mobj = re.match(self._VALID_URL, url)
+        video_id = mobj.group('id')
+        video_type = mobj.group('type')
         cookies = self._get_cookies(url)
-        webpage, urlh = self._download_webpage_handle(url, video_id)
-        st = urlh.headers.get('x-origin-date')
-        app_state = self._parse_json(self._search_regex(
-            r'<script>window\.APP_STATE\s*=\s*({.+?})</script>',
-            webpage, 'app state'), video_id)
-        video_data = {}
-        getters = list(
-            lambda x, k=k: x['initialState']['content%s' % k]['content']
-            for k in ('Data', 'Detail')
-        )
-        for v in app_state.values():
-            content = try_get(v, getters, dict)
-            if content and content.get('contentId') == video_id:
-                video_data = content
-                break
+        if video_type == 'movies':
+            video_data = self._call_api('o/v1/movie/detail', video_id)['body']['results']['item']
+        elif video_type == 'sports':
+            video_data = self._call_api('o/v1/match/detail', video_id)['body']['results']['item']
+        elif video_type == 'episode' or video_type == 'tv':
+            video_data = self._call_api('o/v1/episode/detail', video_id)['body']['results']['item']
+        else:
+            video_data = self._call_api('o/v1/content/detail', video_id)['body']['results']['item']
 
         title = video_data['title']
 
         if not self.get_param('allow_unplayable_formats') and video_data.get('drmProtected'):
             raise ExtractorError('This video is DRM protected.', expected=True)
-
-        headers = {'Referer': url}
+        headers = {'Referer': 'https://www.hotstar.com/in'}
         formats = []
+        subs = {}
         geo_restricted = False
         # change to v2 in the future
-        playback_sets = self._call_api_v2('play/v1/playback', video_id, st=st, cookies=cookies)['playBackSets']
+        playback_sets = self._call_api_v2('play/v1/playback', video_id, cookies=cookies)['playBackSets']
```

Comment:
Doesn't this re-introduce #396?

Reply:
As far as i can tell, OP's system time was diff from HS and therefore he was getting errors. This seems like an external issue to me but still i'll try to see if i can get st from API.
If it's necessary, lemme know and I'll make a web request to `hotstar.com` and get `st`

Reply:
I would normally treat it as an external issue too. But hotstar gives a 403 even with a timestamp that is just a few seconds ahead (behind is fine). And there was pretty much no additional cost to taking the timestamp from the server. But if it takes an additional request and defeat the purpose of the PR, then it may be better to just let it be.

Reply:
The purpose of the PR is mainly to not deal with their redirecting service block, since i've now hardcoded the url, it shouldn't have much headroom. If I were aware of #396 before, i would have not removed `st`.